### PR TITLE
feature: support spring boot actuator

### LINKED ISSUE #4671: Since the `seata-server` already supports `springboot`, can you support the `spring-boot-starter-actuator` so that it can be monitored as a service
## Why you need it?
 Is your feature request related to a problem? Please describe in details

## How it could be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.

### PR DESCRIPTION
**Pull request by l81893521**
### Ⅰ. Describe what this PR did
Support Spring boot actuator(Only support check database connection yet)

Check Url: http://localhost:7091/actuator/health (need authentication)
image

### Ⅱ. Does this pull request fix one issue?
fixes #4671

### CODE REVIEW

File: server/src/main/java/io/seata/server/health/DataBaseHealthIndicator.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+@ConditionalOnExpression("#{'db'.equals('${sessionMode}')}")
+public class DataBaseHealthIndicator implements HealthIndicator {
+
+    protected static final Configuration CONFIG = ConfigurationFactory.getInstance();
+
+    @Override
+    public Health health() {
+
+        String datasourceType = CONFIG.getConfig(ConfigurationKeys.STORE_DB_DATASOURCE_TYPE);
+        DataSource logStoreDataSource = EnhancedServiceLoader.load(DataSourceProvider.class, datasourceType).provide();
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/health/DataBaseHealthIndicator.java:47]: 为什么是个局部变量,还有就是为何不连redis模式一起做了?

Comment (l81893521) [reply server/src/main/java/io/seata/server/health/DataBaseHealthIndicator.java:46]: fixed

Comment (l81893521) [reply server/src/main/java/io/seata/server/health/DataBaseHealthIndicator.java:46]: 因为除了redis还有nacos等各种第三方组件需要检测，所以计划分开做

**Review by funky-eyes — COMMENTED** [review]
Non-transactional features should not be called features, and versions under 2.x will only accept bugfixes, so please change your commit branch to 2.x.

File: server/src/main/java/io/seata/server/health/DataBaseHealthIndicator.java (hunk 1 above)
Comment (funky-eyes) [reply server/src/main/java/io/seata/server/health/DataBaseHealthIndicator.java:46]: > 因为除了redis还有nacos等各种第三方组件需要检测，所以计划分开做

Please submit the split plan as an issue, as one of the next tasks of the community.

**Pull request closed without merging**